docs/api_docs.rst

```diff
@@ -0,0 +1,304 @@
+
+Inputs
+
+* Reference image (numpy.ndarray)
+
+Outputs
+
+* Masks matrix of type ``np.ndarray`` with shape [nMasks x Height x Width].
+
+-----------------------------------------
+Image Occlusion via Perturbation Masks
+-----------------------------------------
+
+A common intermediate step in this process is applying the generated perturbation masks to imagery to produce occluded images. We provide two utility functions as baseline implementation to perform this step. One is the version that performs the transformation as a batch operation, ``xaitk_saliency.utils.masking.occlude_image_batch``. The other, ``xaitk_saliency.utils.masking.occlude_image_streaming``, performs the transformation in a streaming method with optional parallelization in that streaming. While the batch version is simpler and in many cases the faster of the two versions, the streaming version may be more applicable to large image masks or when a great deal of masks are being input, where in such cases the batch version would exceed available memory.
+
+----------------------------
+Saliency Heatmap Generation
+----------------------------
+
+These interfaces comprise a family of siblings that all perform a similar transformation, but requiring different standard inputs. There is no standard to rule them all without being so abstract that it would break the concept of interface abstraction, or the ability to substitute any arbitrary implementations of the interface without interrupting successful execution. Each interface is intended to handle different black-box outputs from different algorithmic categories. In the future, as additional algorithmic categories are identified for which saliency map generation is applicable, additional interfaces may be defined and added to this initial repertoire.
+
+Interface: GenerateDescriptorSimilaritySaliency
+-----------------------------------------------
+.. autoclass:: xaitk_saliency.interfaces.gen_descriptor_sim_sal.GenerateDescriptorSimilaritySaliency
+   :members:
+   :private-members:
+
+This interface proposes that implementations require externally generated feature-vectors for two reference images between which we are trying to discern the feature-space saliency. This also requires the feature-vectors for perturbed images as well as the masks of the perturbations as would be output from a ``PerturbImage`` implementation. We expect perturbations to be relative to the second reference image feature-vector.
```

Comment:
Any additional "narrative" components I imagine we will want to be above the auto-generated API stuff. That said, a lot of this is redundant now with the autoclass rendered text drawn directly from the class' doc-strings. E.g. the "Method", "Input" and "Output" info below is strictly redundant, thus we can remove them now.

I would suggest double checking the content here with the content provided from the class and only retain what's left that is unique.

Reply:
Yes I did notice that redundancy, I will review and revise.